src/main/java/org/dependencytrack/resources/v1/BomResource.java

```diff
@@ -463,4 +471,25 @@ private File validateAndStoreBom(final byte[] bomBytes, final Project project) t
         return tmpFile;
     }
 
+    private static void createWorkflowSteps(QueryManager qm, UUID token) {
+        WorkflowState consumptionState = new WorkflowState();
+        consumptionState.setToken(token);
+        consumptionState.setStep(WorkflowStep.BOM_CONSUMPTION);
+        consumptionState.setStatus(WorkflowStatus.PENDING);
+        WorkflowState parent = qm.persist(consumptionState);
+
+        WorkflowState processingState = new WorkflowState();
+        processingState.setParent(parent);
+        processingState.setToken(token);
+        processingState.setStep(WorkflowStep.BOM_PROCESSING);
+        processingState.setStatus(WorkflowStatus.PENDING);
+        WorkflowState processingParent = qm.persist(processingState);
+
+        WorkflowState vulnAnalysisState = new WorkflowState();
+        vulnAnalysisState.setParent(processingParent);
+        vulnAnalysisState.setToken(token);
+        vulnAnalysisState.setStep(WorkflowStep.VULN_ANALYSIS);
+        vulnAnalysisState.setStatus(WorkflowStatus.PENDING);
+        qm.persist(vulnAnalysisState);
```

Comment:
Would be good to have this done in a single DB transaction, so that either all steps are created, or none.